packages/plugin-ext/src/plugin/plugin-manager.ts

```diff
@@ -139,6 +145,8 @@ export class PluginManagerExtImpl implements PluginManagerExt, PluginManager {
         if (this.host.loadTests) {
             return this.host.loadTests();
         }
+
+        this.fireOnDidChange();
```

Comment:
Are you assured that we need to fire that event every time when we run plugins?
Does VSCode have same behaviour?

Reply:
I have to fire the event once the list of plugins has changed. In the `$init` function we update the list of plugins. That's why I decided to fire the event at the end of function.


Another option is to fire the event once just after the line below:
```
 this.registry.set(plugin.model.id, plugin);
```

Reply:
So VSCode also fired that event when extension system started?

Reply:
Accordingly to the doc:
```
/**		 
 * An event which fires when `extensions.all` changes. This can happen when extensions are
 * installed, uninstalled, enabled or disabled.
 */
```

VS Code fires the event once `_initialize` [1] function is invoked. Here they do nothing but fill the extension Map.
[1] https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/services/extensions/common/extensionDescriptionRegistry.ts#L29


In our implementation `extensions.all` [1] returns values from `this.registry` field [2]
[1] https://github.com/theia-ide/theia/blob/ab/extOnDidChange/packages/plugin-ext/src/plugin/plugin-manager.ts#L195
[2]  https://github.com/theia-ide/theia/blob/ab/extOnDidChange/packages/plugin-ext/src/plugin/plugin-manager.ts#L64

So, once I modify `this.registry` I have to fire the event. So, maybe it is better to fire event just after I modified `this.registry` field.

Reply:
I've updated the PR accordingly.

Reply:
@akosyakov do you have any concerns?